ErgoScript-Developer-Course/batch-0/MGpai/HW#2.md

```diff
@@ -0,0 +1,29 @@
+
+# Homework 2: Recap basic ErgoScript and programming
+
+```
+{
+  val state: Boolean = true //Boolean data type
+  val oneErg: Long = 1000L* 1000L * 1000L // Long data type
+  val ergColl: Coll[Long] = Coll(oneErg, 2*oneErg, 3*oneErg) // Collection data type
+  val moreErg: Coll[Long] = ergColl.append(10*oneErg)
+  val getColl : Int = ergColl(1)
+
+  if(state && (getColl == 2*oneErg)) { // If else
+    val byte: Byte = 1 // Byte data type
+    val pairs: (Byte, Long) = (byte, oneErg) // Pairs data type
+    val mapByteToErg: Coll[(Byte, Long)] = Coll((byte, oneErg), (byte, 3*oneErg))
+    // Arbitrary data type
+    sigmaProp(1==1) // SigmaProp
```

Comment:
Might be a bit of confusion about this from what we went over in class. But this is actually not possible. Remember, that only the end of your contract may return `sigmaProps`. Right now this `if` statement is right in the middle, and there is no `else` statement that also returns a `sigmaProp`. This means the contract would fail to compile.

What you should get out of this: If you're `if` statement returns a `sigmaProp`, it MUST have a matching `else` statement that also returns a `sigmaProp`. This is just the nature of functional programming. Also, anything made after and outside of the `if` and `else` statements would either not be evaluated, or would just throw another compiling error.